### ISSUE TITLE
CDF value does not always approach 1

### ISSUE THREAD

**Issue by drewoldag** [LSSTDESC/qp_flexzboost#9]
@eacharles noted that for examples in the notebook there are cases where the CDF approaches 0.8 or 1.2 for example. 

It's possible that the output from Flexcode post processing with bump threshold removal and peak sharpening is causing some trouble, but it's just as likely that there's a bug in my code. 

If it's Flexcode output, then I'll need to adapt it in qp_flexzboost.

**Comment by drewoldag** [comment]
Some things to check:
- Does the documentation of Flexcode happens to say anything about this?
- Does increasing the number of x values affect the output? Currently using 100 data points by default, what about 300, for instance?

**Comment by drewoldag** [comment]
Flexcode calls a function `normalize` here: https://github.com/lee-group-cmu/FlexCode/blob/master/src/flexcode/basis_functions.py#L214 that theoretically normalizes the PDF to integrate to 1. 

However, after the normalization, bump thresholding and sharpening happens, so clearly the output from FlexZBoost is no longer a true PDF once that happens. 

Perhaps we can use the `normalize` function from Flexcode a second time after the evaluate function is called in qp_flexzboost?

**Comment by drewoldag** [comment]
Experimenting in a new notebook. I set the bump_threshold = sharpen_alpha = None. As I increased the number of grid points from 100 to 30k the `np.max(cdf)` changes from 0.910563 to 0.999702. This is with no additional normalization. 

Using reasonable, non-None, values for bump_threshold=0.1 and sharpen_alpha=1.2. (These values were returned by the model produced in the [rail estimation demo](https://github.com/LSSTDESC/RAIL/blob/main/examples/estimation/RAIL_estimation_demo.ipynb) notebook) and repeating the experiment with grid points from 100 to 30k, the `np.max(cdf)` value changes from 0.799002 to 0.999348. 


Repeating the experiment for x grid with 30k points, and looking at a sample of 10 outputs from Flexcode:
bump_threshold=0.1, sharpen_threshold=1.2
`np.max(cdf)`'s
```
0.99959889
0.99918146
0.99943427
1.00003012
1.00001153
0.99999637
0.99934788
1.00003235
0.99977569
1.00003405
```

Repeating with bump_threshold=sharpen_threshold=None
`np.max(cdf)`'s
```
0.99973481
0.99955924
0.99961722
0.99989446
0.9998857
0.99992206
0.99970216
0.99992448
0.99987105
0.9999034
```
